server/py/services/api/api/endpoints/pipelines.py

```diff
@@ -145,23 +152,116 @@ async def retry_pipeline(
     auth_info: mlrun.common.schemas.AuthInfo = fastapi.Depends(
         framework.api.deps.authenticate_request
     ),
+    submit_mode: str = fastapi.Query(
+        mlrun_constants.RetryMode.rerun, alias="submit-mode"
+    ),
+    db_session: Session = fastapi.Depends(framework.api.deps.get_db_session),
+    client_version: typing.Optional[str] = fastapi.Header(
+        None, alias=mlrun.common.schemas.HeaderNames.client_version
+    ),
 ):
+    project: mlrun.common.schemas.ProjectOut = (
+        await fastapi.concurrency.run_in_threadpool(
+            framework.utils.singletons.project_member.get_project_member().get_project,
+            db_session=db_session,
+            name=project,
+            leader_session=auth_info.session,
+        )
+    )
+
+    # check permission CREATE pipeline
     await (
         framework.utils.auth.verifier.AuthVerifier().query_project_resource_permissions(
-            mlrun.common.schemas.AuthorizationResourceTypes.pipeline,
-            project,
-            run_id,
-            mlrun.common.schemas.AuthorizationAction.create,
-            auth_info,
+            resource_type=mlrun.common.schemas.AuthorizationResourceTypes.pipeline,
+            project_name=project.metadata.name,
+            resource_name=run_id,
+            action=mlrun.common.schemas.AuthorizationAction.create,
+            auth_info=auth_info,
         )
     )
-    run_id = await fastapi.concurrency.run_in_threadpool(
-        services.api.crud.Pipelines().retry_pipeline,
-        run_id,
+
+    original_runner = await fastapi.concurrency.run_in_threadpool(
+        services.api.crud.Pipelines().find_original_workflow_run,
```

Comment:
```suggestion
        services.api.crud.Pipelines().get_original_workflow_run,
```

better naming convention